src/views/catalog/CreateFromInstanceTypes/utils/utils.ts

```diff
@@ -97,36 +84,38 @@ export const generateVM = (
       },
       instancetype: {
         // inferFromVolume: `${virtualmachineName}-disk`,
-        name: instanceTypeName || bootableVolume?.metadata?.labels?.[DEFAULT_INSTANCETYPE_LABEL],
+        name:
+          selectedInstanceType?.name ||
+          selectedBootableVolume?.metadata?.labels?.[DEFAULT_INSTANCETYPE_LABEL],
       },
       preference: {
         // inferFromVolume: `${virtualmachineName}-disk`,
-        name: bootableVolume?.metadata?.labels?.[DEFAULT_PREFERENCE_LABEL],
+        name: selectedBootableVolume?.metadata?.labels?.[DEFAULT_PREFERENCE_LABEL],
       },
       dataVolumeTemplates: [
         {
           metadata: {
             name: `${virtualmachineName}-volume`,
           },
           spec: {
-            ...(isBootableVolumePVCKind(bootableVolume)
+            ...(isBootableVolumePVCKind(selectedBootableVolume)
               ? {
                   source: {
                     pvc: {
-                      name: bootableVolume?.metadata?.name,
-                      namespace: bootableVolume?.metadata?.namespace,
+                      name: selectedBootableVolume?.metadata?.name,
+                      namespace: selectedBootableVolume?.metadata?.namespace,
```

Comment:
You can also use `getNamespace`.

Reply:
even better, instead of repeating this 2 times I created an object that uses both getName and getNamespace :) 
Thanks!